### DISCUSSION TITLE
How to use TopoSeries?

### DISCUSSION THREAD

**Discussion by SuzanneSoy** [DeepSOIC/Lattice2#56] [General]
I'm trying to attach copies of the sketch `Feature` (the trapezium at the top) to some of the edges of another sketch (`Path` just below). I'm expecting to obtain the shape on the right (in my actual use case, I'd only select a subset of the edges). The `Feature` sketch has 4 degrees of freedom, which allow placing both its endpoints and it will stretch as needed. The use case is to add gluing tabs for a model made out of paper.

![image](https://user-images.githubusercontent.com/299872/103236234-3f12e380-493c-11eb-9554-f52fa11b8907.png)

I tried to click `TopoSeries` after selecting the `Feature` and the `Edges of Path` (made with the `Downgrade to Edges` tool). I also tried to click `TopoSeries` after selecting the `Feature` and the `Array from Edges of Path` (made with `Downgrade to Edges` → `Array from shape: internal placements` with `orient mode = child.Edge` and `Translate mode = child.Vertex`)

In both cases, I get this message:

> Error: 
> In supplied links, nothing to loop over compounds was found. Key-by-key errors follow: 

How is the TopoSeries tool supposed to be used? I tried looking for documentation but couldn't find any example.

Here's the FreeCAD document in the screenshot: [test_lattice2_toposeries.FCStd.zip](https://github.com/DeepSOIC/Lattice2/files/5748453/test_lattice2_toposeries.FCStd.zip)

I just discoved Lattice2 and it seems very useful so far :)

**Comment by DeepSOIC** [comment]
Hi! I'm sorry I still haven't written a tutorial for this tool, it is not easy to figure out.

The sketch "feature" must be based on "Edges of Path" (one or several; preferably on Edge1). Then to do the toposeries, you select "Feature" and "Edges of Path" and click "Toposeries". That's it.

However, since TopoSeries actually modifies the external-geometry links of the sketch in order to compute the occurrences, doing it the straightforward way usually results in a collapsed sketch at some step. Attaching the sketch to the linked edge can grealy improve the odds of solver success, which is what i did in the example file below. TopoSeries will loop the attachment link as well.

[test_lattice2_toposeries_done.FCStd.zip](https://github.com/DeepSOIC/Lattice2/files/5749020/test_lattice2_toposeries_done.FCStd.zip)

"Array from Edges of Path" in your file is not needed and is not helpful for toposeries.
